File: C/Algorithms/Sorting/Selection_Sort.c
#include<stdio.h>
#include<stdlib.h>

void selectionSort(int a[], int n);
void print(int a[], int n);

void main() 
{   
    int length, i;
    printf("Welcome to the Selection Sort Algorithm!!\n");
    printf("You can sort the data elements here, I will help you in that\n");
    printf("So lets start with sorting integer datatypes\n");
    printf("So let me know the how much integer datatypes you want me to sort:\n");
    scanf("%d", &length);

    // Creating an array using calloc i.e. array for the user defined length
    int *array = (int *) calloc (length, sizeof(int));

    // Accepting the data elements for the array
    printf("Now you can enter the integers:\n");
    for ( i = 0; i < length; i++)
    {
        scanf("%d", &array[i]);
    }

    // Implemention the Selection Sort (in Ascending Order)
    selectionSort(array, length);
    
    // Printing the Sorted elements
    print(array, length);
}

void selectionSort(int a[], int n)
{
    int i, j, minimunIndex, temp;

    printf("Just started to sort Using the Selection Sort Algorithm\n");

    for ( i = 0; i < n; i++)
    {
        minimunIndex = i;

        for ( j = i + 1; j < n; j++)
        {
            if (a[minimunIndex] > a[j])
                minimunIndex = j;    
        }
        temp = a[i];
        a[i] = a[minimunIndex];
        a[minimunIndex] = temp;
    }

    printf("Ended Sorting using Selection Sort\n");
}

void print(int a[], int n)
{
    int i;
    printf("So now Printing the Array below:\n");

    for ( i = 0; i < n; i++)
    {
        printf("%c%4d%c\n", 221, a[i], 222);
    }    
}

File: SelectionSort.java
import java.util.Scanner;

public class SelectionSort {

	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		System.out.println("Enter how many number of elements you want in the a");
int n=sc.nextInt();
int a[]=new int[n];
for(int i=0;i<a.length;i++)
{
	a[i]=sc.nextInt();
}
System.out.println("Your array");
for(int l:a)
{
	System.out.print(l+" ");
}

System.out.println();

System.out.println("After selection sorting");
int m[]=arr(a);
for(int b:m)
{
	System.out.print(b+" ");
}
	}
static int[] arr (int a[])
{
	for(int i=0;i<a.length-1;i++)
	{ int minindex=i;
		for(int j=i;j<a.length;j++)
		{
			if(a[j]<a[minindex])
			{
				minindex=j;
			}
		int temp=a[i];
		a[i]=a[minindex];
		a[minindex]=temp;
		}
	}
return a;
}
}
